src/zcash/address/zip32.h

```diff
@@ -64,109 +57,20 @@ class HDSeed {
-        zip339_phrase_to_seed(language, mnemonic.c_str(), (uint8_t (*)[64])seed.data());
-    }
-
-public:
-    MnemonicSeed(Language languageIn, SecureString mnemonicIn): language(languageIn), mnemonic(mnemonicIn) {
-        SetSeedFromMnemonic();
-    }
-
-    /**
-     * Randomly generate a new mnemonic seed. A SLIP-44 coin type is required to make it possible
-     * to check that the generated seed can produce valid transparent and unified addresses at account
-     * numbers 0x7FFFFFFF and 0x00 respectively.
-     */
-    static MnemonicSeed Random(uint32_t bip44CoinType, Language language = English, size_t entropyLen = 32);
-
-    static std::string LanguageName(Language language) {
-        switch (language) {
-            case English:
-                return "English";
-            case SimplifiedChinese:
-                return "Simplified Chinese";
-            case TraditionalChinese:
-                return "Traditional Chinese";
-            case Czech:
-                return "Czech";
-            case French:
-                return "French";
-            case Italian:
-                return "Italian";
-            case Japanese:
-                return "Japanese";
-            case Korean:
-                return "Korean";
-            case Portuguese:
-                return "Portuguese";
-            case Spanish:
-                return "Spanish";
-            default:
-                return "INVALID";
-        }
-    }
-
-    ADD_SERIALIZE_METHODS;
-
-    template <typename Stream, typename Operation>
-    inline void SerializationOp(Stream& s, Operation ser_action) {
-        if (ser_action.ForRead()) {
-            uint32_t language0;
-
-            READWRITE(language0);
-            READWRITE(mnemonic);
-            language = (Language) language0;
-            SetSeedFromMnemonic();
-        } else {
-            uint32_t language0 = (uint32_t) language;
-
-            READWRITE(language0);
-            READWRITE(mnemonic);
-        }
-   }
-
-    template <typename Stream>
-    static MnemonicSeed Read(Stream& stream) {
-        MnemonicSeed seed;
-        stream >> seed;
-        return seed;
-    }
-
-    const Language GetLanguage() const {
-        return language;
-    }
-
-    const SecureString& GetMnemonic() const {
-        return mnemonic;
-    }
-
-    friend bool operator==(const MnemonicSeed& a, const MnemonicSeed& b)
-    {
-        return a.seed == b.seed;
-    }
-
-    friend bool operator!=(const MnemonicSeed& a, const MnemonicSeed& b)
-    {
-        return !(a == b);
-    }
-};
-
 // This is not part of ZIP 32, but is here because it's linked to the HD seed.
 uint256 ovkForShieldingFromTaddr(HDSeed& seed);
 
 namespace libzcash {
 
 typedef uint32_t AccountId;
 
+/**
+ * The account identifier used for HD derivation of
+ * transparent and Sapling addresses via the legacy
+ * `getnewaddress` and `z_getnewaddress` code paths,
+ */
+const AccountId ZCASH_LEGACY_ACCOUNT = HARDENED_KEY_LIMIT - 1;
```

Comment:
The `uint32_t` to `AccountId` change here is another non-move change that I missed in my previous review. This is a correct fix to the change made in #5357.